kernel_gateway/gatewayapp.py

```diff
@@ -221,17 +240,31 @@ def init_webapp(self):
         '''
         # Redefine handlers off the base_url path
         handlers = []
-        for handler in (
-            default_kernel_handlers + 
-            default_kernelspec_handlers +
-            default_base_handlers
-        ):
-            # Create a new handler pattern rooted at the base_url
-            pattern = url_path_join(self.base_url, handler[0])
-            # Some handlers take args, so retain those in addition to the
-            # handler class ref
-            new_handler = tuple([pattern] + list(handler[1:]))
-            handlers.append(new_handler)
+        if self.api:
+            kernel_id = self.kernel_manager.start_kernel()
+            kernel_client = self.kernel_manager.get_kernel(kernel_id).client()
+            # discover the notebook endpoints and their implementations
+            endpoints = self.kernel_manager.endpoints()
+            sorted_endpoints = self.kernel_manager.sorted_endpoints()
+            # append tuples for the notebook's API endpoints
+            for uri in endpoints:
+                parameterized_path = parameterize_path(uri)
+                print('Registering uri {}'.format(parameterized_path))
```

Comment:
Leftover print. Probably do want logging here though so that the person bringing things up can see what's exposed.